### PR TITLE
refactor(python,rust): Rename `Expr.list` to `implode`

### PR DESCRIPTION
**Pull request by stinodego**
Related to #7529

We want to rename the `arr` namespace to `list`, but this is slightly harder than initially thought.

I want to nicely deprecate `Expr.list`, which I have done in this PR. But I cannot introduce the `Expr.list` property while this method still exists.

What should we do? 🤔 

Also, I'd like to get @alexander-beedie 's opinion on this one; always good insights on naming stuff :) and maybe you have some dark magic to make this work?

### CODE REVIEW

**Comment by ritchie46** [comment]
Hmm... Let's start with the deprecation of this method. We can do another breaking release in a few weeks and then deprecate the `arr` namespace.

File: py-polars/src/lazy/dsl.rs [hunk 1]
```diff
@@ -187,7 +187,7 @@ impl PyExpr {
     pub fn last(&self) -> PyExpr {
         self.clone().inner.last().into()
     }
-    pub fn list(&self) -> PyExpr {
+    pub fn implode(&self) -> PyExpr {
         self.clone().inner.list().into()
```
Comment (ritchie46) [inline py-polars/src/lazy/dsl.rs:191]: I think we must follow the `list` trace all the way down. It should be called `implode` on the rust side as well now.

Comment (stinodego) [reply py-polars/src/lazy/dsl.rs:191]: Will do 👍

**Comment by stinodego** [comment]
All right. And should the new `implode` be available on the functions level? e.g. should we offer `pl.implode("name")`?

Not sure if we want to offer all expressions in this way or only certain ones?

**Comment by ritchie46** [comment]
> All right. And should the new `implode` be available on the functions level? e.g. should we offer `pl.implode("name")`?
> 
> Not sure if we want to offer all expressions in this way or only certain ones?

Hmm.. Might be. I am not sure if this function does more magic than this. If it is a mere alias for `pl.col().implode()`, I think we should rename it as well.

**Comment by alexander-beedie** [comment]
Hmm, I would have been tempted to go with `array` instead of `list`, to maintain the `.arr-` muscle-memory and avoid namespace confusion/conflation with Python's native list type? Though that would require the additional large-scale change of `pl.List` to `pl.Array` to keep the associated dtype name in sync, so I can see why `list` is the pragmatic choice 😅 

We should try and work out how to repoint docs too - I'm sure we've been leaving a trail of broken links into the API docs in our wake recently, and this will do the same to an entire namespace.

**Comment by alexander-beedie** [comment]
Aha! And if you need black magic, you've come to the right place... that would be for having two different active `list` attributes? (One pointing at `implode`, and one leading to the new version of `arr`). 

I think that can actually be done 😅 Should be possible to temporarily support it by adding `__call__` to the new list namespace, passing it through to `implode`.

`expr.list()  →  implode`
`expr.list.method(...)  →  namespace methods`

**Comment by stinodego** [comment]
Let's just do it in two steps, like Ritchie suggested. There's nothing wrong with doing another breaking release sometime soon, in my opinion.

**Comment by stinodego** [comment]
There was a lot to update, but it's ready!

**Comment by ritchie46** [comment]
Thanks a lot @stinodego. Looks great.

**Pull request merged**